lms/djangoapps/grades/rest_api/v1/tests/test_gradebook_views.py

```diff
@@ -1031,6 +1032,184 @@ def test_page_size_parameter(self, page_size):
+                self.assertEqual(expected_results, actual_data['results'])
+                self.assertEqual(actual_data['total_users_count'], num_enrollments)
+                self.assertEqual(actual_data['filtered_users_count'], 2)
+
+    @ddt.data(
+        ['login_staff', 4],
+        ['login_course_admin', 5],
+        ['login_course_staff', 5]
+    )
+    @ddt.unpack
+    def test_filter_course_grade_min_and_max(self, login_method, num_enrollments):
+        with patch('lms.djangoapps.grades.course_grade_factory.CourseGradeFactory.read') as mock_grade:
+            # even though we're creating actual PersistentCourseGrades below, we still need
+            # mocked subsection grades
+            mock_grade.side_effect = [
+                self.mock_course_grade(self.program_student, passed=True, percent=0.75),
+            ]
+
+            PersistentCourseGrade(
+                user_id=self.student.id,
+                course_id=self.course_key,
+                percent_grade=0.85
+            ).save()
+            PersistentCourseGrade(
+                user_id=self.other_student.id,
+                course_id=self.course_key,
+                percent_grade=0.45
+            ).save()
+            PersistentCourseGrade(
+                user_id=self.program_student.id,
+                course_id=self.course_key,
+                percent_grade=0.75
+            ).save()
+
+            with override_waffle_flag(self.waffle_flag, active=True):
+                getattr(self, login_method)()
+                resp = self.client.get(
+                    self.get_url(course_key=self.course.id) + '?course_grade_min=50&course_grade_max=80'
+                )
+
+                expected_results = [
+                    OrderedDict([
+                        ('user_id', self.program_student.id),
+                        ('username', self.program_student.username),
+                        ('email', ''),
+                        ('external_user_key', 'program_user_key_0'),
+                        ('percent', 0.75),
+                        ('section_breakdown', self.expected_subsection_grades()),
+                    ]),
+                ]
+
+                self.assertEqual(status.HTTP_200_OK, resp.status_code)
+                actual_data = dict(resp.data)
+                self.assertEqual(expected_results, actual_data['results'])
+                self.assertEqual(actual_data['total_users_count'], num_enrollments)
+                self.assertEqual(actual_data['filtered_users_count'], 1)
+
+    @ddt.data(
+        ['login_staff', 4],
+        ['login_course_admin', 5],
+        ['login_course_staff', 5]
+    )
+    @ddt.unpack
+    def test_filter_course_grade_absent(self, login_method, num_enrollments):
+        with patch('lms.djangoapps.grades.course_grade_factory.CourseGradeFactory.read') as mock_grade:
+            # even though we're creating actual PersistentCourseGrades below, we still need
+            # mocked subsection grades
```

Comment:
I got myself a little turned around while determining what this comment meant. I think it's because it's so surprising to me that something called `mock_course_grade` would mock subsection-level grades in addition to course-level ones. I think in this context the name of the method is mirroring the name of a kind of object, rather than the semantics of what's being done, and we might be able to change it to a more comprehensive name. Open to alternatives/maybe a longer comment?

(But to drive home the extent of my confusion: I pulled up the file, found the definition, and managed still to gloss over the fact that it was doing what you'd said in this comment somehow. It took a second look after I'd already half-written a comment about it for me to get myself turned right again.)

Reply:
I agree. In fact, the thing I was stumped on for a while yesterday was the fact that this function does both, which is why I added this comment. That should have been a clue to me to address the confusion. I had removed all references to the mock method in my tests, since I was creating `PersistentSubsectionGrade` objects and figured I didn't need to mock the course grades anymore, but then the `subsection_breakdown` returned by the API no longer matched the expected values since those were no longer being mocked. 

I'm struggling to think of a good name. `mock_course_and_subsection_grades`? Do you have any suggestions?

Reply:
I think `mock_course_and_subsection_grades` is great!